File: Src/task_one.cpp
//
// Created by FindBin on 2024/10/17.
//

#include <iostream>
#include <string>
#include <map>
#include <thread>

void fn_task_one() {
    std::string input_str{};
    std::cin >> input_str;
    std::map<std::string, std::string> command_map{
            {"name",  "FBResistor"},
            {"class", "MPS-Class"},
            {"emil",  "1578557118@qq.com"},
            {"phone", "13620464128"},
            {"exit",  "bye"}
    };
    for (;;) {
        auto result = command_map.find(input_str);
        if (result != command_map.end()) {
            std::cout << result->second << std::endl;
            if (result->first == std::string{"exit"}) {
                break;
            }
        } else {
            std::cout << "try again" << std::endl;
        }
        std::this_thread::sleep_for(std::chrono::milliseconds(10));
    }
}

File: rust/src/task_one.rs
use std::collections::HashMap;
use std::io;
use std::thread;
use std::time::Duration;

pub fn final_call() {
    let command_map: HashMap<String, String> = HashMap::from([
        ("name".to_string(), "FBResistor".to_string()),
        ("class".to_string(), "MPS-Class".to_string()),
        ("emil".to_string(), "1578557118@qq.com".to_string()),
        ("phone".to_string(), "13620464128".to_string()),
        ("exit".to_string(), "bye".to_string()),
    ]);

    loop {
        let mut input_str = String::new();
        io::stdin().read_line(&mut input_str).expect("error");
        let input_str = input_str.trim();
        if let Some(res) = command_map.get(input_str) {
            println!("{}", res);
            if input_str == "exit" {
                break;
            }
        } else {
            println!("try again");
        }

        thread::sleep(Duration::from_millis(10));
    }
}
